source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/ViewController.m

```diff
@@ -247,30 +252,33 @@ - (void)update:(NSString *) jsonStr
     {
         ACRView *ad = renderResult.view;
         ad.mediaDelegate = self;
-        if(self.curView)
-            [self.curView removeFromSuperview];
-
-        self.curView = ad;
-
-        [_scrView addSubview:self.curView];
-        UIView *view = self.curView;
-        view.translatesAutoresizingMaskIntoConstraints = NO;
-            
-        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0].active = YES;
-        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0].active = YES;
-        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeLeading multiplier:1.0 constant:3].active = YES;
-        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeTrailing multiplier:1.0 constant:0].active = YES;
+        //        if(self.curView)
+        //            [self.curView removeFromSuperview];
+        
+        //        self.curView = ad;
+        //[self.scrViewA removeAllObjects];
+        [self.scrViewA addObject:ad];
+        [self.scrView reloadData];
+        //        [_scrView addSubview:self.curView];
+        //        UIView *view = self.curView;
+        //        view.translatesAutoresizingMaskIntoConstraints = NO;
+        //
+        //        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0].active = YES;
+        //        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0].active = YES;
+        //        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeLeading multiplier:1.0 constant:3].active = YES;
+        //        [NSLayoutConstraint constraintWithItem:view attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:_scrView attribute:NSLayoutAttributeTrailing multiplier:1.0 constant:0].active = YES;
```

Comment:
we probably shouldn't be checking in commented-out code :)

Reply:
and below

---
In reply to: [263089605](https://github.com/Microsoft/AdaptiveCards/pull/2505#discussion_r263089605) [](ancestors = 263089605)